benchmark.go

```diff
@@ -188,12 +213,57 @@ func runBenchmark(out output, logger *zap.Logger, allOpts Options, resolved reso
 		zap.Time("startTime", start),
 	)
 
+    if opts.Format == "json" || opts.Format == "JSON" {
+        // Calling JSON output helper method
+        outputJSON(opts, overall, out, goMaxProcs, numConns, total)
+    } else {
+        // Calling plaintext output helper method
+        outputPlaintext(opts, overall, out, goMaxProcs, numConns, total)
+	}
+}
+
+// JSON output helper method
+func outputJSON(opts BenchmarkOptions, overall *benchmarkState, out output, goMaxProcs int, numConns int, total time.Duration) {
+	// Print out errors
 	overall.printErrors(out)
-	overall.printLatencies(out)
+	// Create Latencies, BenchmarkParameters, and Summary struct
+    latencyValues := overall.getLatencies(out)
+    benchmarkParams := BenchmarkParameters{CPUs: goMaxProcs, Connections: numConns, Concurrency: opts.Concurrency, MaxRequests: opts.MaxRequests, MaxDuration: opts.MaxDuration.String(), MaxRPS: opts.RPS}
+	latenciesOutput := Latencies{P5000: latencyValues[0], P9000: latencyValues[1], P9500: latencyValues[2], P9900: latencyValues[3], P9990: latencyValues[4], P9995: latencyValues[5], P1000: latencyValues[6]}
+	summary := Summary{ElapsedTime: (total / time.Millisecond * time.Millisecond).String(), TotalRequests: overall.totalRequests, RPS: float64(overall.totalRequests)/total.Seconds()}
+	// Create output struct and format into JSON
+    benchmarkOutput := BenchmarkOutput{BenchmarkParameters: benchmarkParams, Latencies: latenciesOutput, Summary: summary}
+    functionOutput, err := json.MarshalIndent(&benchmarkOutput, "", "  ")
```

Comment:
nits: avoid naked arguments, it's recommended to add a comment on what those means, e.g. 
```
json.MarshalIndent(&benchmarkOutput, "" /* prefix */, "  " /* indent */)
```
ref: https://github.com/uber-go/guide/blob/master/style.md#avoid-naked-parameters